melodies_monet/plots/ozone_sonder_plots.py

```diff
@@ -0,0 +1,230 @@
+    df_height = pd.DataFrame({
+         'station':['Boulder, Colorado','Huntsville, Alabama','University of Rhode Island','Trinidad Head, California'],
+         'height':[1.743,0.203,0.021,0.046]
+         })
+    if altitude_method[0] == 'ground level':
+        height_value = df_height.loc[df_height['station']==station_name[0]].values[0][1]
+        ALT_gl = ALT_sl - height_value
+
+    #set default figure size
+    if fig_dict is not None:
+        f,ax = plt.subplots(**fig_dict)
+    else:
+        f,ax = plt.subplots(figsize=(8,8))
+
+    #set default text size
+    def_text = dict(fontsize=20)
+    if text_dict is not None:
+        text_kwargs = {**def_text, **text_dict}
+    else:
+        text_kwargs = def_text
+
+    #set color style
+    obs_dict = dict(color='k', linestyle='-',marker='*', linewidth=2.0, markersize=5.)
+    if plot_dict is not None:
+        plot_kwargs = {**obs_dict, **plot_dict}
+        plot_kwargs.pop('column')
+    else:
+        plot_kwargs = obs_dict
+
+    #Define characteristics of boxplot.
+    boxprops = {'edgecolor': 'k', 'facecolor': 'w','linewidth': 1.5}
+    lineprops = {'color': 'k', 'linewidth': 1.5}
+    boxplot_kwargs = {'boxprops': boxprops, 'medianprops': lineprops,
+                  'whiskerprops':  lineprops, 'capprops': lineprops,
+                  'widths':0.3, 'patch_artist': True,'showfliers': False
+#                  'zorder':0,'showfliers': False
+#                  'showmeans': False,'notch'= False,'showfliers': False
+                  }
+    
+    #Make plot
+    #get position list
+    position_list_1 = altitude_threshold_list[:-1]
+    position_list_2 = altitude_threshold_list[1:]
+    position_list_mid = [(position_list_1[i]+position_list_2[i])/2 for i in range(len(position_list_1))]
+
+    if altitude_method[0] =='sea level':
+        output_list_obs = split_by_threshold(O3_OBS,ALT_sl,altitude_threshold_list)
+        bplot_obs=ax.boxplot(output_list_obs,vert = False,**boxplot_kwargs,positions=position_list_mid)
+        #add a line plot
+        ax.plot(O3_OBS,ALT_sl,'-k*',label = 'OBS:'+model_name_list[0])
+
+        for i in range(len(O3_MODEL_ALL)):
+            output_list_model = split_by_threshold(O3_MODEL_ALL[i],ALT_sl,altitude_threshold_list)
+            bplot_model=ax.boxplot(output_list_model,vert = False,**boxplot_kwargs,positions=position_list_mid)
+            #add a line plot
+            ax.plot(O3_MODEL_ALL[i],ALT_sl,'-*',color=fill_color_list[i+1],label = 'MODEL:'+model_name_list[i+1])
+
+            colors=[fill_color_list[i+1]]*(len(altitude_threshold_list)-1)
+            for patch,color in zip(bplot_model['boxes'],colors):
+                patch.set_edgecolor(color)
+
+        plt.title('Comparison at '+str(station_name[0])+' on '+str(release_time)+' UTC')
+        plt.legend()
+        plt.ylim(altitude_range[0],altitude_range[1])
+        plt.xlim(ozone_range[0],ozone_range[1])
+        plt.ylabel('ALT-sea level (km)')
+        plt.xlabel('O3 (ppbv)')
```

Comment:
Once you code the vmin_plot and vmax_plot in you will also be able to access the ylabel_plot and could use that here too to make this code more universal and not just for Ozone. And you could expand your code to look at temperature for example very easily with some small updates like this.

Reply:
this is done!